### ISSUE TITLE
[uikit-storefront] Create package!

### ISSUE THREAD

**Issue by bmuenzenmeyer** [pattern-lab/patternlab-node#896]
### Background

Required reading: @bradfrost wrote http://bradfrost.com/blog/post/the-workshop-and-the-storefront/ outlining a vision for separate spaces for separate design system / component library use cases and audiences.

He wrote http://bradfrost.github.io/style-guide-guide/ as a means to take Pattern Lab workshops and transform them into storefronts.

![image](https://user-images.githubusercontent.com/298435/42720508-95866818-86ed-11e8-96ba-16ae9ae71d8f.png)

### UIKits

#840 Introduced UIKits to Pattern Lab, allowing users to build more than one front end from a single set of source patterns. `styleguidekit-assets-default` and `styleguidekit-mustache-default` became `uikit-workshop` - the default UI we have come to know as Pattern Lab.

I created https://github.com/pattern-lab/uikit-bare as a jumping off point for others to take a stripped down version of `uikit-workshop` and build a new UI. Brad and I are working to further whittle that down, but this issue is about building out `uikit-storefront`

**Done When:** the screenshot above can be created by importing `@pattern-lab/uikit-storefront`

### Considerations

* [Page support](https://github.com/pattern-lab/patternlab-node/issues/788) is likely a necessary component of this solution
* UIKits currently have an `excludedTags` and `excludedPatternStates` feature. We may need a white-list approach too.

**Comment by bmuenzenmeyer** [comment]
@bradfrost this is an umbrella issue for our larger efforts. We can tag-team efforts after uikit-bare gets a bit more clean.

Hype!

**Comment by bradfrost** [comment]
@taylor-benson 

> Is this proposal to have style-guide-guide incorporated into the same repo as the pattern-lab project? 

Yeah, that's the general spirit of it. The idea is to be able to create multiple builds of PL from the same source files. The workshop is the default view of PL, but the idea is to be able to create the storefront view from the same source code, which can (in theory at least) re-create the Style Guide Guide inside of Pattern Lab.

>  I know this would probably take much work but we definitely appreciate the effort @bmuenzenmeyer and @bradfrost have put in. 

Yeah it's definitely an effort, and is something that I need to carve out time to work on. So I wouldn't hold your breath waiting for that to happen immediately. I'd recommend starting with the Style Guide Guide, and thankfully because the content all markdown it should be easy-ish to migrate everything over when the storefront feature is ready. 

Thanks!

**Comment by sghoweri** [comment]
Virtually all of the UIKit refactor work I’ve been chipping away at relates to platform-level changes we need to get us to this future state without sacrificing the flexibility teams need.

As @bradfrost mention, there’s still plenty of work to be done however we are still making progress!

**Comment by Hydraner** [comment]
I think in the meanwhile it would still be interesting to just use the uikit-workshop for the storefront usecase. I did some work on making uikit's reusable for multiple frontends. Just wanted to reference some issues in here for people who find themself in the same situation searching for an easy way to have a storefront without having a dedicated uikit for it:
#1136
#901
#844
